pyiceberg/catalog/glue.py

```diff
@@ -305,7 +308,18 @@ def __init__(self, name: str, **properties: Any):
             aws_secret_access_key=get_first_property_value(properties, GLUE_SECRET_ACCESS_KEY, AWS_SECRET_ACCESS_KEY),
             aws_session_token=get_first_property_value(properties, GLUE_SESSION_TOKEN, AWS_SESSION_TOKEN),
         )
-        self.glue: GlueClient = session.client("glue", endpoint_url=properties.get(GLUE_CATALOG_ENDPOINT))
+        self.glue: GlueClient = session.client(
+            "glue",
+            endpoint_url=properties.get(GLUE_CATALOG_ENDPOINT),
+            config=Config(
+                retries={
+                    "max_attempts": get_first_property_value(properties, GLUE_MAX_RETRIES)
+                    if get_first_property_value(properties, GLUE_MAX_RETRIES)
+                    else MAX_RETRIES,
+                    "mode": "standard",
```

Comment:
Should we also make the mode configurable? :)

Reply:
I would rather not have that as an option. Standard mode is a more mature implementation compared to the legacy one.Adaptive mode is still considered 'experimental' which is not something I'd use in production.

Reply:
+1 to making it configurable just to future-proof. 
Still set `standard` as the default, but can be overridden if necessary

Reply:
Ok, I'll make the mode configurable

Reply:
@mark-major We're almost there, are you interested in pushing this over the finish line?

Reply:
I have pushed the changes, sorry for taking this long...